### DISCUSSION TITLE
Axum match pattern failing to nest and route service

### DISCUSSION THREAD

**Discussion by spikeHollow** [tokio-rs/axum#2573] [Q&A]
### Summary

Here's the code
```rust
let router = Router::new()
            .nest_service("/", /* capture all "/*" routes and forward it one service */*/some())
            .route_service("/", /* just check this / and forward it another service */another())
```
I expected it to happen like I described, nut, its just throwing route already registered error.
Initially, I discussed this in rust [forum](https://users.rust-lang.org/t/axum-match-pattern-failing-to-nest-and-route-service/106420).
However I tried adding a wildcard "/*key" like they said and I got this error.
```
thread 'main' panicked at ~/.cargo/registry/src/index.crates.io-6f17d22bba15001f/axum-0.7.4/src/routing/path_router.rs:211:20:
Invalid route: nested routes cannot contain wildcards (*)
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```
How to solve this issue?

### axum version

0.7.4

**Comment by davidpdrsn** [comment]
That is by design. `.nest_service("/")` will match all requests, including `/`. So the routes are overlapping.

**Comment by jplatte** [comment]
Instead of nesting at `/` you can use `router.fallback_service()`.
